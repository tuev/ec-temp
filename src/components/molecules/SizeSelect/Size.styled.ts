import styled from 'styled-components'

export const SizeSelectWrapper = styled.div``

export const SizeItemWrapper = styled.div`
  margin: 0 4px;
  display: inline-block;

  button {
    border-radius: 4px;
    min-height: 28px !important;
    min-width: 55px !important;
    line-height: 20px;
  }

  &:first-child {
    margin-left: 0;
  }

  &:last-child {
    margin-right: 0;
  }
`
