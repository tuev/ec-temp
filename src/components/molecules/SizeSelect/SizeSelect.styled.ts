import styled from 'styled-components'
import { InputLabel } from '@material-ui/core'
import { createMuiTheme } from '@material-ui/core/styles'

export const SizeSelectWrapper = styled.div``

export const SizeItemWrapper = styled.div`
  margin: 0 4px;
  display: inline-block;

  button {
    border-radius: 4px;
    min-height: 28px !important;
    min-width: 55px !important;
    line-height: 20px;
  }

  &:first-child {
    margin-left: 0;
  }

  &:last-child {
    margin-right: 0;
  }
`

export const SizeInputLabel = styled(InputLabel)`
  /* background-color: white; */
`

export const theme = createMuiTheme({
  props: {
    MuiInputBase: {
      style: {
        backgroundColor: 'white',
        borderRadius: 32,
        width: 120,
      },
    },
    MuiOutlinedInput: {
      margin: 'dense',
    },
  },
})
