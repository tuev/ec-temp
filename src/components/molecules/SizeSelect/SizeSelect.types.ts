import { LayoutProps } from 'styled-system'

export type SIZE_PARAMS = 'M' | 'S' | 'XS' | 'L' | 'XL' | 'XXL' | 'XXXL'

type HandleOnChange = (size: SIZE_PARAMS) => unknown

export interface ISizeSelect {
  sizes?: SIZE_PARAMS[]
  value: SIZE_PARAMS
  onChange?: HandleOnChange
  type?: 'radio' | 'select'
}

export type BasicSizeSelectProps = LayoutProps & ISizeSelect
