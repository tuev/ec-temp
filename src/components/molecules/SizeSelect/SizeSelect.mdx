import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './SizeSelect.stories.tsx'
import SizeSelect from './SizeSelect.tsx'

<Story name="basic" />
<Story name="example" />
<Story name="dropdown" />
