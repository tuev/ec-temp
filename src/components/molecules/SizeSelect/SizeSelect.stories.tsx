import React, { useCallback, useState } from 'react'

import SizeSelect from './SizeSelect'
import { SIZE_PARAMS } from './Size.type'

export default {
  component: SizeSelect,
  includeStories: ['basic', 'example'],
  title: 'SizeSelect',
}

export const basic = () => <SizeSelect size={['M', 'XL', 'L']} value={'L'} />

export const example = () => <SizeSelectExample />

const SizeSelectExample = () => {
  const sizes: SIZE_PARAMS = ['M', 'L', 'XL', 'XXL']
  const [size, toggleSize] = useState<SIZE_PARAMS>('L')
  const handleToggleSize = useCallback(sizeCode => toggleSize(sizeCode), [])

  return <SizeSelect sizes={sizes} value={size} onChange={handleToggleSize} />
}
