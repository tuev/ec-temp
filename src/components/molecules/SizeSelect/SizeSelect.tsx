import React, { FC, useCallback, useState, useEffect } from 'react'
import { BasicSizeSelectProps, SIZE_PARAMS } from './SizeSelect.types'
import Button from 'components/atoms/Button'
import {
  SizeSelectWrapper,
  SizeItemWrapper,
  SizeInputLabel,
  theme,
} from './SizeSelect.styled'
import { FormControl, Select, MenuItem } from '@material-ui/core'
import { ThemeProvider } from '@material-ui/core/styles'
import Typography from 'components/atoms/Typography'

const SizeSelect: FC<BasicSizeSelectProps> = props => {
  const { sizes = [], value, onChange, type = 'radio' } = props
  const [size, onSizeChange] = useState<SIZE_PARAMS>(sizes[0] || '')
  const handleSizeOnChange = useCallback(
    // todo: need to make typecheck for hof
    (sizeValue: SIZE_PARAMS) => () => onChange && onChange(sizeValue),
    [onChange]
  )

  const handleSelect = useCallback(
    (e: React.ChangeEvent<{ value: unknown }>) =>
      onSizeChange(e.target.value as SIZE_PARAMS),
    [onSizeChange]
  )

  useEffect(() => {
    // tslint:disable-next-line: no-unused-expression
    onChange && onChange(size)
  }, [size, onChange])

  return type === 'radio' ? (
    <SizeSelectWrapper>
      {sizes.map(sizeItem => (
        <SizeItemWrapper key={sizeItem}>
          <Button
            mx="4px"
            color={size === sizeItem ? 'secondary' : 'default'}
            onClick={handleSizeOnChange(sizeItem)}
            size="small"
            data-testid={`size-filter-${sizeItem}`}
          >
            {sizeItem}
          </Button>
        </SizeItemWrapper>
      ))}
    </SizeSelectWrapper>
  ) : (
    <ThemeProvider theme={theme}>
      <FormControl variant="outlined">
        <SizeInputLabel>Size</SizeInputLabel>
        <Select
          value={size}
          onChange={handleSelect}
          data-testid="size-filter-select"
        >
          {sizes.map(sizeItem => (
            <MenuItem value={sizeItem} key={sizeItem}>
              <Typography data-testid={`size-filter-${sizeItem}`}>
                {sizeItem}
              </Typography>
            </MenuItem>
          ))}
        </Select>
      </FormControl>
    </ThemeProvider>
  )
}

export default SizeSelect
