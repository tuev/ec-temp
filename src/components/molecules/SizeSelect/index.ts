import SizeSelect from './SizeSelect'

export default SizeSelect
