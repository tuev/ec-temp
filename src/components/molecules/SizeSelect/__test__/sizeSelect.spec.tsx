import React from 'react'

import {
  render,
  fireEvent,
  getByTestId,
  queryAllByTestId,
} from '@testing-library/react'
import SizeSelect from '..'
import { SIZE_PARAMS } from '../SizeSelect.types'

// NOTE: https://codesandbox.io/s/pknk1v4mmx for testing style-components
// * using: data-testid on component and getByTestId return by 'render' of react-testing-library

describe('App size select', () => {
  it('Test get default size select no params ', () => {
    // test snapshot
    const wrapper = render(<SizeSelect />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Test get default size select type select no params ', () => {
    // test snapshot
    const wrapper = render(<SizeSelect type="select" />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Test get default size select ', () => {
    const sizes: SIZE_PARAMS[] = ['M', 'L', 'XS', 'XXL', 'XL']
    // test snapshot
    const wrapper = render(<SizeSelect sizes={sizes} value="XS" />)
    expect(wrapper).toMatchSnapshot()
  })

  it('Test get default size select type dropdown ', () => {
    const sizes: SIZE_PARAMS[] = ['M', 'L', 'XS', 'XXL', 'XL']
    // test snapshot
    const wrapper = render(
      <SizeSelect sizes={sizes} value="XS" type="select" />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('Test get default size filter with actions ', () => {
    let value: SIZE_PARAMS = 'M'
    const sizes: SIZE_PARAMS[] = ['M', 'L', 'XS', 'XXL', 'XL']
    const onChange = size => (value = size)
    // test snapshot
    const wrapper = render(
      <SizeSelect sizes={sizes} value={value} onChange={onChange} />
    )

    expect(wrapper).toMatchSnapshot()

    sizes.forEach(size => {
      const sizeRadio = wrapper.getByTestId(`size-filter-${size}`)
      fireEvent.click(sizeRadio)
      expect(value).toBe(size)
    })
  })

  it('Test get default size filter select with actions ', () => {
    let value: SIZE_PARAMS = 'M'
    const sizes: SIZE_PARAMS[] = ['M', 'L', 'XS', 'XXL', 'XL']
    const onChange = size => (value = size)
    // test snapshot
    const wrapper = render(
      <SizeSelect
        sizes={sizes}
        value={value}
        onChange={onChange}
        type="select"
      />
    )

    expect(wrapper).toMatchSnapshot()
    const selectSizeComp = wrapper.getByTestId('size-filter-select')

    sizes.forEach(size => {
      fireEvent.click(selectSizeComp)
      const sizeRadio = wrapper.getByTestId(`size-filter-${size}`)
      fireEvent.click(sizeRadio)
      console.log(size, 'sizes')
      expect(value).toBe(size)
    })
  })
})
