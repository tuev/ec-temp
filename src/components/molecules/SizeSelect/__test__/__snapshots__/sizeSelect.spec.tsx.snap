// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App size select Test get default size filter with actions  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c1 {
  margin-left: 4px;
  margin-right: 4px;
  min-width: 90px !important;
  min-height: 36px !important;
  font-size: 12px;
  border-radius: 24px;
}

.c0 {
  margin: 0 4px;
  display: inline-block;
}

.c0 button {
  border-radius: 4px;
  min-height: 28px !important;
  min-width: 55px !important;
  line-height: 20px;
}

.c0:first-child {
  margin-left: 0;
}

.c0:last-child {
  margin-right: 0;
}

<div>
      <div
        class=""
      >
        <div
          class="c0"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained c1 MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
            data-testid="size-filter-M"
            mx="4px"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              M
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="c0"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained c1 MuiButton-containedSizeSmall MuiButton-sizeSmall"
            data-testid="size-filter-L"
            mx="4px"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              L
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="c0"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained c1 MuiButton-containedSizeSmall MuiButton-sizeSmall"
            data-testid="size-filter-XS"
            mx="4px"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              XS
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="c0"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained c1 MuiButton-containedSizeSmall MuiButton-sizeSmall"
            data-testid="size-filter-XXL"
            mx="4px"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              XXL
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="c0"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained c1 MuiButton-containedSizeSmall MuiButton-sizeSmall"
            data-testid="size-filter-XL"
            mx="4px"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              XL
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bwzfXH uTxCW"
    >
      <div
        class="sc-htpNat jtYCUb"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bdVaJa gKhRMU MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
          data-testid="size-filter-M"
          mx="4px"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            M
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="sc-htpNat jtYCUb"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bdVaJa gKhRMU MuiButton-containedSizeSmall MuiButton-sizeSmall"
          data-testid="size-filter-L"
          mx="4px"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            L
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="sc-htpNat jtYCUb"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bdVaJa gKhRMU MuiButton-containedSizeSmall MuiButton-sizeSmall"
          data-testid="size-filter-XS"
          mx="4px"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            XS
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="sc-htpNat jtYCUb"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bdVaJa gKhRMU MuiButton-containedSizeSmall MuiButton-sizeSmall"
          data-testid="size-filter-XXL"
          mx="4px"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            XXL
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="sc-htpNat jtYCUb"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bdVaJa gKhRMU MuiButton-containedSizeSmall MuiButton-sizeSmall"
          data-testid="size-filter-XL"
          mx="4px"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            XL
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App size select Test get default size select  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    .c1 {
  margin-left: 4px;
  margin-right: 4px;
  min-width: 90px !important;
  min-height: 36px !important;
  font-size: 12px;
  border-radius: 24px;
}

.c0 {
  margin: 0 4px;
  display: inline-block;
}

.c0 button {
  border-radius: 4px;
  min-height: 28px !important;
  min-width: 55px !important;
  line-height: 20px;
}

.c0:first-child {
  margin-left: 0;
}

.c0:last-child {
  margin-right: 0;
}

<div>
      <div
        class=""
      >
        <div
          class="c0"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained c1 MuiButton-containedSizeSmall MuiButton-sizeSmall"
            data-testid="size-filter-M"
            mx="4px"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              M
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="c0"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained c1 MuiButton-containedSizeSmall MuiButton-sizeSmall"
            data-testid="size-filter-L"
            mx="4px"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              L
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="c0"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained c1 MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
            data-testid="size-filter-XS"
            mx="4px"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              XS
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="c0"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained c1 MuiButton-containedSizeSmall MuiButton-sizeSmall"
            data-testid="size-filter-XXL"
            mx="4px"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              XXL
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="c0"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained c1 MuiButton-containedSizeSmall MuiButton-sizeSmall"
            data-testid="size-filter-XL"
            mx="4px"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              XL
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bwzfXH uTxCW"
    >
      <div
        class="sc-htpNat jtYCUb"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bdVaJa gKhRMU MuiButton-containedSizeSmall MuiButton-sizeSmall"
          data-testid="size-filter-M"
          mx="4px"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            M
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="sc-htpNat jtYCUb"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bdVaJa gKhRMU MuiButton-containedSizeSmall MuiButton-sizeSmall"
          data-testid="size-filter-L"
          mx="4px"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            L
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="sc-htpNat jtYCUb"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bdVaJa gKhRMU MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
          data-testid="size-filter-XS"
          mx="4px"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            XS
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="sc-htpNat jtYCUb"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bdVaJa gKhRMU MuiButton-containedSizeSmall MuiButton-sizeSmall"
          data-testid="size-filter-XXL"
          mx="4px"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            XXL
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="sc-htpNat jtYCUb"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-bdVaJa gKhRMU MuiButton-containedSizeSmall MuiButton-sizeSmall"
          data-testid="size-filter-XL"
          mx="4px"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            XL
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`App size select Test get default size select no params  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=""
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bwzfXH uTxCW"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
